README.md
# Plugin: WP German Formal - WP Deutsch formelle Anrede

(Eigene) Formelle deutsche Übersetzungen für WordPress Core laden und anzeigen lassen - für die Bereiche Global, Admin und Netzwerkverwaltung.

## Beschreibung

Lädt anstelle der "offiziellen deutschen Sprachdateien" (die mit der WordPress DE-Edition kommen) eigene deutsche Übersetzungen für WordPress - übersetzt, geplegt und herausgegeben von David Decker von DECKERWEB. Dies betrifft die folgenden Bereiche bzw. Sprachdateien: Global, Adminbereich sowie die Netzwerkverwaltung in Multisite.

* de_DE.mo = der globale Bereich, d.h. Strings, die überall vorkommen können, im Frontend, wie im Adminbereich
* admin-de_DE.mo = Strings, die nur im Adminbereich vorkommen
* network-admin-de_DE.mo = Strings, die nur in der Netzwerkverwaltung vorkommen (Network Admin in Multisite)

Diese neuen Sprachdateien basieren auf den "offiziellen Sprachdateien" aus der DE-Edition, in der Sie-Anrede. Sie wurden jedoch (nahezu) komplett überarbeitet und teilweise stark verändert.

**Grundlegendes**

 1. [Übersetzungsprinzipien](https://github.com/deckerweb/wp-german-formal/wiki/%C3%9Cbersetzungsprinzipien)
 2. [Begriffstabelle/ -Vergleich - extern, via Google Docs](https://docs.google.com/spreadsheet/ccc?key=0AsdlEocpfc1CdFRjNzVQRkpYZ2lVRnozbmJXcEpQcmc&usp=sharing)
 3. [WordPress Glossar - extern, via Google Docs](https://docs.google.com/spreadsheet/ccc?key=0AsdlEocpfc1CdHdTaEYxdDBTZU4tWEFtOTN0bnZKUlE&usp=sharing)

## Screenshots

Derzeit keine. (Im Moment nicht relevant.)

## Anfoderungen
* WordPress Version 3.7 oder höher (damit auch PHP 5.2.4 oder höher!)
* Ihre WordPress-Installation ist bereits auf deutsch eingestellt ('WPLANG' auf 'de_DE' in der wp-config.php)

## Installation

### Hochladen

1. Die neueste Version hier von [den GitHub.com Releases](https://github.com/deckerweb/wp-german-formal/releases) herunterladen (die "zip"-Option wählen).
2. Zur Admin-Seite __Plugins -> Installieren__ gehen und dort den Tab __Hochladen__ wählen.
3. Das ZIP-Paket direkt hochladen.
4. Nach dem Hochladeprozess auf __Aktivieren__ klicken.

### Manuell

1. Die neueste Version hier von [den GitHub.com Releases](https://github.com/deckerweb/wp-german-formal/releases) herunterladen (die "zip"-Option wählen).
2. Das heruntergeladene Archiv entpacken (extrahieren).
3. Kopieren Sie den entpackten Ordner in Ihr `/wp-content/plugins/` Verzeichnis (in der Regel via FTP).
4. Gehen Sie auf die Admin-Seite "Plugins", wählen Sie das entsprechende plugin und klicken __Aktivieren__.

### Git

Via git können Sie zu Ihrem Verzeichnis `/wp-content/plugins/` gehen und dieses Repositorium klonen:

`git clone git@github.com:deckerweb/wp-german-formal.git`

Gehen Sie auf die Admin-Seite "Plugins", wählen Sie das entsprechende plugin und klicken __Aktivieren__.

## Sprachdatei- und Plugin-Aktualisierungen (Updates)

Neue Versionen der Sprachdateien, z.B. bei einer neuen WordPress-Version, werden über eine neue Version des Plugins veröffentlicht.

Dieses Plugin unterstützt das Updater-Plugin für GitHub: [GitHub Updater](https://github.com/afragen/github-updater). Wenn Sie dieses also zuerst installieren, wird "WP German Formal" genauso aktualisierbar, wie jedes andere Plugin.

## Verwendung des Plugins

Das Plugin hat keine Einstellungen. Es wird einfach aktiviert und verrichtet lautlos seinen Dienst. :)

Für die Zukunft ist eventuell eine Unterseite im Adminbereich mit Informationen bzw. einigen Einstellungen geplant. Wann das geschehen wird, ist derzeit völlig offen.

Für Multisite-Installationen wird eine Netzwerk-weite Aktivierung empfohlen.

Derzeit werden bis zu 3 (drei!) verschiedene Hauptpersionen von WordPress unterstützt: 3.8 (aktuell), davor 3.7 sowie zukünftig 3.9. Mehr wie drei solcher Versionen werden nicht unterstützt! Diese Funktionalität ist sehr komfortabel zum Testen sowie bei Aktualisierungen von Installationen! :-)

**Zusatzfunktion:**
Blocker-Konstante, um Aktualisierung von Übersetzungen (Sprachpakete - Language Packs) zu unterbinden, auch für WordPress Core. Betrifft nur die Lokale 'de_DE'. Rein optional über Hinzufügen der folgenden Konstante in der `wp-config.php` aktivierbar:
```php
/** Plugin: WP German Formal */
define( 'WPGF_BLOCK_DE_LANGUAGE_PACKS', TRUE );
```

## Lizenz

GPL-2.0+
siehe: [LICENSE Datei](https://github.com/deckerweb/wp-german-formal/blob/master/LICENSE)

## Credits

Entwickelt und herausgegeben von [David Decker - DECKERWEB](http://deckerweb.de/twitter)  
Copyright (c) 2013 [David Decker - DECKERWEB](http://deckerweb.de/)

CHANGES.md
### Changelog for WP German Formal

#### 1.0.8 (2013-12-17)
* DE:
 * Sprachdateien für 3 WordPress-Hauptversionen nun möglich (z.B. für 3.7, 3.8 und 3.9) - kann nützlich sein für Aktualisierungen, zum Testen usw.
 * Jetzt inklusive "Blocker"-Filter/ -Funktionen, um die Aktualisierung deutscher Sprachpakete (Language Packs) zu unterbinden, auch für WordPress Core (nur für deutsche Lokale 'de_DE'). Aktivierbar über Konstante in `wp-config.php`.
 * Übersetzungen auf aktuellstem Stand, 17.12.2013
* EN:
 * Language packs for 3 major versions of WordPress are now possible (for example: 3.7, 3.8 and 3.9) - could be useful for Updates, for testing etc.
 * Now including "blocker" filters/ function to deactivate updates of German language packs, including WordPress Core - only for German locale (de_DE); activation optional via constant in `wp-config.php`.
 * Translations of current state, as of 2013-12-17

#### 1.0.7 (2013-12-11)
* DE: Übersetzungen auf aktuellstem Stand, 10.12.2013
* EN: Translations of current state, as of 2013-12-10

#### 1.0.6 (2013-12-09)
* DE:
 * Neu: Jetzt mit den Übersetzungen für alle Standard Themes (Default Themes) von 2010-2014, d.h. für "Twenty Ten" bis "Twenty Fourteen"! (Achtung: Bei "Twenty Fourteen" handelt es sich noch um eine Entwicklungsversion - daher sind die Übersetzungen nicht final!)
 * Sprachdateien nun für WordPress Versionen 3.7 und 3.8 beiliegend. (Achtung: Bei 3.8 handelt es sich noch um eine Entwicklungsversion - daher sind die Übersetzungen nicht final!)
 * Übersetzungen auf aktuellstem Stand, 09.12.2013
* EN:
 * New: Now with translations for all default themes, too, for 2010-2014 so for "Twenty Ten" up to "Twenty Fourteen". (Note: "Twenty Fourteen" is still in development so translations are not final!)
 * Now with packaged language files for WordPress versions 3.7 and 3.8. (Note: WP 3.8 is still in development so translations are not final!)
 * Translations of current state, as of 2013-12-09

#### 1.0.5 (2013-11-20)
* DE:
 * Sprachdateien für 2 WordPress-Hauptversionen nun möglich (z.B. für 3.7 und 3.8) - kann nützlich sein für Aktualisierungen, zum Testen usw.
 * Übersetzungen auf aktuellstem Stand, 20.11.2013
 * Readme.txt Datei hinzugefügt (sinnvoll für WordPress-Ökosystem).
* EN:
 * Language packs for 2 major versions of WordPress are now possible (for example: 3.7 and 3.8) - could be useful for Updates, for testing etc.
 * Translations of current state, as of 2013-11-20
 * Added readme.txt file (useful for WordPress ecosystem).

#### 1.0.2 (2013-11-18)
* DE:
 * Erste öffentliche stabile Version
 * Übersetzungen auf aktuellstem Stand, 18.11.2013
* EN:
 * Initial public stable release
 * Translations of current state, as of 2013-11-18

#### 0.9.0 (2013-11-18)
* DE: Erste öffentliche Betaversion
* EN: Initial public beta release

### Upgrade Notice

#### 1.0.8
DE: Sprachdateien aktualisiert. // EN: Updated language packs.

#### 1.0.7
DE: Sprachdateien aktualisiert. // EN: Updated language packs.

#### 1.0.6
DE: Sprachdateien für WordPress 3.7 und 3.8 sowie alle Standard Themes 2010-2014. // EN: Language packs for WordPress 3.7 and 3.8 plus all default themes 2010-2014.

#### 1.0.5
DE: Sprachdateien für 2 WordPress-Hauptversionen möglich. Readme.txt hinzugefügt. // EN: Language packs for 2 major versions of WordPress possible. Added readme.txt file.

#### 1.0.2
DE: Erste öffentliche stabile Version. // EN: First public stable release.

#### 0.9.0
DE: Erste öffentliche Betaversion. // EN: First public beta.

includes/wpgf-blocker.php
<?php
/**
 * Helper functions for for setting blockers to language pack updating.
 *
 * @package    WP German Formal
 * @subpackage Language Packs
 * @author     David Decker - DECKERWEB
 * @copyright  Copyright (c) 2013, David Decker - DECKERWEB
 * @license    http://www.opensource.org/licenses/gpl-license.php GPL-2.0+
 * @link       http://genesisthemes.de/en/wp-plugins/wp-german-formal/
 * @link       http://deckerweb.de/twitter
 *
 * @since      1.0.8
 */

/**
 * Exit if accessed directly
 *
 * @since 1.0.8
 */
if ( ! defined( 'WPINC' ) ) {
	exit( 'Sorry, you are not allowed to access this file directly.' );
}


/** Deactivate Language Pack Updates - for de_DE locale: */
if ( 'de_DE' == get_locale() ) {

	add_filter( 'auto_update_translation', '__return_false' );

}  // end if

/** WordPress Core updates: always use default locale */
add_filter( 'core_version_check_locale', '__return_empty_string' );


add_action( 'core_upgrade_preamble', 'ddw_wpgf_blocker_info', 11 );
/**
 * User message on "update-core.php" admin screen about updates when using a
 *    localized version of WordPress.
 *
 * Hooked into: 'core_upgrade_preamble' / priority: 11
 *
 * @since 1.0.8
 */
function ddw_wpgf_blocker_info() {

	$update_info_message = sprintf(
		'<br /><h3>%s</h3><p>%s</p><p>%s</p>',
		'&rarr; ' . esc_attr__( 'Important info for "WP German Formal" plugin:', 'wp-german-formal' ),
		esc_attr__( 'You are currently have enabled the "WP German Formal" plugin to let load your translations for WordPress Core (global, admin, network admin). That is, generally speaking, perfectly fine. :) So thank you, first!', 'wp-german-formal' ),
		sprintf(
			esc_attr__( 'You have currently defined our helper constant %s in your %s file. Therefore updates for language packs are deactivated for the German locale %s. Also, your WordPress Core install will download/ update no WordPress Core language packs (for WordPress itself), so it will always use the default version (that is locale %s). To change that behavior, just remove or uncomment the helper constant.', 'wp-german-formal' ),
			'<code>WPGF_BLOCK_DE_LANGUAGE_PACKS</code>',
			'<code>wp-config.php</code>',
			'<code>de_DE</code>',
			'<code>en_US</code>'
		)
	);

	?>

		<div id="wpgf-update-info" class="description"><?php echo $update_info_message; ?></div>

	<?php

}  // end of function ddw_wpgf_blocker_info

includes/wpgf-functions.php
<?php
/**
 * Various helper functions.
 *
 * @package    WP German Formal
 * @subpackage Functions
 * @author     David Decker - DECKERWEB
 * @copyright  Copyright (c) 2013, David Decker - DECKERWEB
 * @license    http://www.opensource.org/licenses/gpl-license.php GPL-2.0+
 * @link       http://genesisthemes.de/en/wp-plugins/wp-german-formal/
 * @link       http://deckerweb.de/twitter
 *
 * @since      1.0.0
 */

/**
 * Exit if accessed directly
 *
 * @since 1.0.0
 */
if ( ! defined( 'WPINC' ) ) {
	exit( 'Sorry, you are not allowed to access this file directly.' );
}


/**
 * User message on "update-core.php" admin screen about updates when using a
 *    localized version of WordPress.
 *
 * Hooked into: 'core_upgrade_preamble' / priority: 11
 *
 * @since 1.0.0
 */
function ddw_wpgf_update_info() {

	$update_info_message = sprintf(
		'<br /><h3>%s</h3><p>%s</p><p>%s</p>',
		'&rarr; ' . esc_attr__( 'Important info for "WP German Formal" plugin:', 'wp-german-formal' ),
		esc_attr__( 'You are currently have enabled the "WP German Formal" plugin to let load your translations for WordPress Core (global, admin, network admin). That is, generally speaking, perfectly fine. :) So thank you, first!', 'wp-german-formal' ),
		esc_attr__( 'However, if you were/ are originally using a localized version of WordPress, let\'s say with German language pack, you can update your WordPress install like normal. It will also update the default (a.k.a. packaged) language packs. Since you have the above mentioned plugin active, the language packs from that plugin will be loaded and displayed. To have the default translations displayed you must deactivate the mentioned plugin.', 'wp-german-formal' )
	);

	?>

		<div id="wpgf-update-info" class="description"><?php echo $update_info_message; ?></div>

	<?php

}  // end of function ddw_wpgf_update_info

wp-german-formal.php
<?php
/**
 * Main plugin file.
 * Load (custom) formal German translations for WordPress Core - Global, Admin
 *    and Network Admin, plus default themes.
 *
 * @package           WP German Formal
 * @author            David Decker
 * @copyright         Copyright (c) 2013, David Decker - DECKERWEB
 * @license           GPL-2.0+
 * @link              http://deckerweb.de/twitter
 *
 * @wordpress-plugin
 * Plugin Name:       WP German Formal
 * Plugin URI:        http://genesisthemes.de/en/wp-plugins/wp-german-formal/
 * Description:       Load (custom) formal German translations for WordPress Core - Global, Admin and Network Admin, plus default themes.
 * Version:           1.0.8
 * Author:            David Decker - DECKERWEB
 * Author URI:        http://deckerweb.de/
 * License:           GPL-2.0+
 * License URI:       http://www.opensource.org/licenses/gpl-license.php
 * Text Domain:       wp-german-formal
 * Domain Path:       /wpgf-languages/
 * GitHub Plugin URI: https://github.com/deckerweb/wp-german-formal
 * GitHub Branch:     master
 *
 * Copyright (c) 2013 David Decker - DECKERWEB
 *
 *     This file is part of WP German Formal,
 *     a plugin for WordPress.
 *
 *     WP German Formal is free software:
 *     You can redistribute it and/or modify it under the terms of the
 *     GNU General Public License as published by the Free Software
 *     Foundation, either version 2 of the License, or (at your option)
 *     any later version.
 *
 *     WP German Formal is distributed in the hope that
 *     it will be useful, but WITHOUT ANY WARRANTY; without even the
 *     implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
 *     PURPOSE. See the GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with WordPress. If not, see <http://www.gnu.org/licenses/>.
 */

/**
 * Exit if accessed directly
 *
 * @since 1.0.0
 */
if ( ! defined( 'WPINC' ) ) {
	exit( 'Sorry, you are not allowed to access this file directly.' );
}


/**
 * Setting constants.
 *
 * @since 1.0.0
 */
/** Set constant for the plugin directory */
define( 'WPGF_PLUGIN_DIR', trailingslashit( dirname( __FILE__ ) ) );

/** Set constant path to the Plugin basename (folder) */
define( 'WPGF_PLUGIN_BASEDIR', trailingslashit( dirname( plugin_basename( __FILE__ ) ) ) );

/** Set constant path to the Plugin URI */
define( 'WPGF_PLUGIN_URI', trailingslashit( plugin_dir_url( __FILE__ ) ) );

/**
 * Helper constant for next major WordPress Version AFTER the current one.
 * CURRENT: 3.8
 *  FUTURE: 3.9
 *
 * @since 1.0.3
 */
define( 'WPGF_WPJUMP_VERSION', '3.8.9999' );

/**
 * Helper constant for previous major WordPress Version BEFORE the current one.
 *   OLDIE: 3.7
 * CURRENT: 3.8
 *
 * @since 1.0.8
 */
define( 'WPGF_WPJUMP_OLDIE_VERSION', '3.7.9999' );


add_action( 'init', 'ddw_wpgf_init', 1 );
/**
 * Load the text domain for translation of the plugin itself.
 * 
 * @since 1.0.0
 *
 * @uses  load_textdomain() Loads translations for given plugin textdomain.
 * @uses  load_plugin_textdomain() Loads translations for given plugin textdomain.
 * @uses  is_admin() Check if we are within admin.
 */
function ddw_wpgf_init() {

	/** If 'wp-admin' include translations plus admin helper functions */
	if ( is_admin() ) {

		/** Set unique textdomain string */
		$wpgf_textdomain = 'wp-german-formal';

		/** The 'plugin_locale' filter is also used by default in load_plugin_textdomain() */
		$plugin_locale = apply_filters( 'plugin_locale', get_locale(), $wpgf_textdomain );

		/** Set filter for WordPress languages directory */
		$wpgf_wp_lang_dir = apply_filters(
			'wpgf_filter_wp_lang_dir',
			trailingslashit( WP_LANG_DIR ) . 'wpgf-languages/' . $wpgf_textdomain . '-' . $plugin_locale . '.mo'
		);

		/** Set filter for plugin languages directory */
		$wpgf_lang_dir = apply_filters( 'wpgf_filter_lang_dir', WPGF_PLUGIN_BASEDIR . 'wpgf-languages' );

		/** Translations: First, look in WordPress' "languages" folder = custom & update-secure! */
		load_textdomain( $wpgf_textdomain, $wpgf_wp_lang_dir );

		/** Translations: Secondly, look in plugin's "languages" folder = default */
		load_plugin_textdomain( $wpgf_textdomain, FALSE, $wpgf_lang_dir );

	}  // end if is_admin() check

	/** Include helper functions */
	include_once( WPGF_PLUGIN_DIR . 'includes/wpgf-functions.php' );

}  // end of function ddw_wpgf_init


/**
 * Helper functions for disabling language pack updates (only if 'de_DE' locale)
 *    and only allow WordPress Core updates with default locale (so no Core
 *    language packs are downloaded/updated).
 *
 * USAGE: Add to your wp-config.php file:
 *        define( 'WPGF_BLOCK_DE_LANGUAGE_PACKS', TRUE );
 *
 * @since 1.0.8
 */
if ( defined( 'WPGF_BLOCK_DE_LANGUAGE_PACKS' ) && WPGF_BLOCK_DE_LANGUAGE_PACKS ) {

	include_once( WPGF_PLUGIN_DIR . 'includes/wpgf-blocker.php' );

} else {

	add_action( 'core_upgrade_preamble', 'ddw_wpgf_update_info', 11 );

}  // end if


/**
 * Helper function for loading our custom translations for WordPress.
 *    It checks first in a subfolder of WP_LANG_DIR, then it falls back to our
 *    plugin subfolder to load the plugin's packaged translations.
 *
 * @since  1.0.0
 *
 * @uses   get_locale()
 * @uses   load_textdomain()
 *
 * @param  string $domain
 * @param  string $key
 *
 * @global string $GLOBALS[ 'wp_version' ]
 */
function ddw_wpgf_load_textdomain( $domain, $key ) {

	/** Get locale setting */
	$locale = get_locale();

	/** Making base slug filterable */
	$custom_slug = 'ddw';  // 'wp-custom';
	$custom_slug = apply_filters( 'wpgf_filter_wplangdir_custom_slug', $custom_slug );

	/** Enable version jumping - for updates, testing and such... :) */
	//$wpjump_slug = ( version_compare( $GLOBALS[ 'wp_version' ], WPGF_WPJUMP_VERSION, '>=' ) ) ? 'future' : '/';
	if ( version_compare( $GLOBALS[ 'wp_version' ], WPGF_WPJUMP_VERSION, '>=' ) ) {

		$wpjump_slug = 'future';

	} elseif ( version_compare( $GLOBALS[ 'wp_version' ], WPGF_WPJUMP_OLDIE_VERSION, '<=' ) ) {

		$wpjump_slug = 'oldies';

	} else {

		$wpjump_slug = '/';

	}  // end if


	/**
	 * First, look with subfolder of WP_LANG_DIR
	 * Example: wp-content/languages/ddw/{future}/admin-de_DE.mo
	 */
	$mofile_wp_lang_dir = trailingslashit( WP_LANG_DIR ) . trailingslashit( esc_attr( $custom_slug ) ) . trailingslashit( $wpjump_slug ) . esc_attr( $key ) . $locale . '.mo';

	/**
	 * Second, fallback to plugin's /pomo/ folder
	 * Pattern: wp-content/plugins/wp-german-formal/wp-pomo/{future}/admin-de_DE.mo
	 */
	$plugindir_slug = apply_filters(
		'wpgf_filter_plugindir_slug',
		trailingslashit( dirname( plugin_basename( __FILE__ ) ) )
	);

	$mofile_plugin_dir = trailingslashit( WP_PLUGIN_DIR ) . $plugindir_slug . 'wp-pomo/' . trailingslashit( $wpjump_slug ) . esc_attr( $key ) . $locale . '.mo';

	/** Loading logic, file check */
	if ( is_readable( $mofile_wp_lang_dir ) ) {

		load_textdomain( esc_attr( $domain ), $mofile_wp_lang_dir );

	} elseif ( is_readable( $mofile_plugin_dir ) ) {

		load_textdomain( esc_attr( $domain ), $mofile_plugin_dir );

	}  // end if

}  // end of function ddw_wpgf_load_textdomain


add_action( 'plugins_loaded', 'ddw_wpgf_load_default_textdomain', 0 );
/**
 * Re-load the 'default' textdomain, use our own, packaged translation files.
 *    All that happens after unloading the original default/ packaged
 *    translations (that come from translate.wordpress.org).
 *
 * NOTE: I am using the approach with unloading and re-loading because the
 *       filters 'load_textdomain_mofile' and 'override_load_textdomain' only
 *       work with one loaded instance of a textdomain, not more (we need
 *       3 times loading of the same textdomain, in global area, admin, plus
 *       network admin).
 *
 * @since 1.0.0
 *
 * @uses  unload_textdomain()
 * @uses  ddw_wpgf_load_textdomain()
 * @uses  is_admin()
 * @uses  is_network_admin()
 */
function ddw_wpgf_load_default_textdomain() {

	/** Unload default files first */
	unload_textdomain( 'default' );

	/** Global: loading */
	ddw_wpgf_load_textdomain( 'default', '' );

	/** Admin */
	if ( is_admin()
		|| ( defined( 'WP_REPAIRING' ) && WP_REPAIRING )
	) {

		ddw_wpgf_load_textdomain( 'default', 'admin-' );

	}  // end if

	/** Network Admin */
	if ( is_network_admin()
		|| ( defined( 'WP_INSTALLING_NETWORK' ) && WP_INSTALLING_NETWORK )
	) {

		ddw_wpgf_load_textdomain( 'default', 'admin-network-' );

	}  // end if

}  // end of function ddw_wp_load_default_textdomain


add_action( 'after_setup_theme', 'ddw_wpgf_load_default_themes_textdomain' );
/**
 * Re-load the textdomain for a default theme, use our own, packaged translation
 *    files. All that happens after unloading the original default/ packaged
 *    translations (that come from translate.wordpress.org).
 *
 * NOTE: See above remarks for ddw_wpgf_load_default_textdomain().
 *
 * @since 1.0.6
 *
 * @uses  wp_get_theme()
 * @uses  unload_textdomain()
 * @uses  ddw_wpgf_load_textdomain()
 */
function ddw_wpgf_load_default_themes_textdomain() {

	$theme = wp_get_theme();

	$theme_domain = esc_attr( $theme->display( 'TextDomain' ) );

	/** 2010-2014 Twenty Ten, Twenty Eleven, Twenty Twelve, Twenty Thirteen, Twenty Fourteen */
	if ( in_array( $theme_domain, array( 'twentyfourteen', 'twentythirteen', 'twentytwelve', 'twentyeleven', 'twentyten' ) ) ) {

		unload_textdomain( $theme_domain );

		ddw_wpgf_load_textdomain( $theme_domain, $theme_domain . '-' );

	}  // end if

}  // end of function